### DISCUSSION TITLE
Creating the notion of a "Customer" within BTCPay

### DISCUSSION THREAD

**Discussion by dstrukt** [btcpayserver/btcpayserver#2630] [Plugins & integrations]
One thing I've been considering after seeing all of the new customer fields being added to the "create invoice" view is the notion of adding a customer, so that the "customer" becomes a reusable, re-selectable entity when filling out an invoice.

It's tedious and frustrating if you have to re-enter this information multiple times on an invoice, and this seems like a reasonable means of addressing that problem.

However, there are potentially privacy concerns with this, but I suppose if that information is already being input and stored (?), would enabling this as a distinct feature be much different?

There's definitely more to think about here, and what the implementation of this might look like more concretely, but I wanted to throw this out there to get some feedback / get some perspective on something I might not be considering?

Anyway, all feedback and thoughts welcome!

**Comment by Kukks** [comment]
This could be interesting as a plugin! Either as a customer management UI or an invoice template UI which could be used to prefill invoices with

**Reply by pavlenex** [reply]
I don't think this should be a plugin, this to me is a core feature. Explicit use-case https://github.com/btcpayserver/btcpayserver/issues/2631

**Reply by dstrukt** [reply]
Hadn't considered it as a plugin, and maybe there's some other information that might sway me...? but as of this moment, I second @pavlenex's thinking as well, core functionality.
